core/src/main/scala/org/apache/spark/ExecutorAllocationClient.scala

```diff
@@ -23,10 +23,16 @@ package org.apache.spark
 private[spark] trait ExecutorAllocationClient {
 
   /**
+   * Express a preference to the cluster manager for a given total number of executors. This can
+   * result in canceling pending requests or filing additional requests.
+   */
+  private[spark] def requestTotalExecutors(numExecutors: Int): Boolean
+
+  /**
    * Request an additional number of executors from the cluster manager.
-   * Return whether the request is acknowledged by the cluster manager.
+   * This is currently only supported in YARN mode. Return whether the request is received.
```

Comment:
If we want to add in the javadocs that it's only supported in YARN mode, we should do it for all the methods here or just move this to the class javadocs. I prefer the latter.